CMakeLists.txt

```diff
@@ -297,8 +297,8 @@ add_subdirectory(misc)
 include(misc/generated/CMakeLists.txt)
 add_subdirectory(boards)
 add_subdirectory(ext)
-add_subdirectory(subsys)
 add_subdirectory(arch)
```

Comment:
Seems like 'arch' should be first before boards, etc.. and we should add a comment about why its first here.